org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -369,23 +400,27 @@ public void beanChangePerformed(BeanEvent<U> evt) {
 				} else {
 					IConsumerProcess<U> process = ref.get();
 					if (process!=null) {
-						process.getBean().setStatus(bean.getStatus());
-						process.getBean().setMessage(bean.getMessage());
-						if (bean.getStatus()==Status.REQUEST_TERMINATE) {
-							processes.remove(bean.getUniqueId());
-							if (process.isPaused()) process.resume();
-							process.terminate();
-						} else if (bean.getStatus()==Status.REQUEST_PAUSE) {
-							process.pause();
-						} else if (bean.getStatus()==Status.REQUEST_RESUME) {
-							process.resume();
-						}
+						manageProcess(process, bean);
 					}
 				}
 			} catch (EventException ne) {
 				logger.error("Internal error, please contact your support representative.", ne);
 			}
 		}
+
+		private void manageProcess(IConsumerProcess<U> process, U bean) throws EventException {
+			process.getBean().setStatus(bean.getStatus());
+			process.getBean().setMessage(bean.getMessage());
+			if (bean.getStatus()==Status.REQUEST_TERMINATE) {
+				processes.remove(bean.getUniqueId());
+				if (process.isPaused()) process.resume();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)